tests/testthat/test-emissions_profile_any_compute_profile_ranking.R

```diff
@@ -62,111 +53,53 @@ test_that("without crucial columns errors gracefully", {
-    tilt_sector = c("a", "a", "a"),
-    tilt_subsector = c("a", "a", "a"),
-    unit = c("a", "a", "a")
-  )
-  isic <- emissions_profile_any_compute_profile_ranking(co2) |>
-    filter((str_length(isic_4digit) %in% c(4, 5)) & (grouped_by %in% c("isic_4digit", "unit_isic_4digit")))
-  expect_equal(unique(isic$profile_ranking), NA_integer_)
+  should_be_na <- out |>
+    filter(is.na(get_column(out, pattern))) |>
+    filter(grepl(pattern, grouped_by))
+  expect_equal(unique(should_be_na$profile_ranking), NA_integer_)
 })
 
-test_that("2 and 3 digit ISICs should be included for ranking
-          all benachmarks other than `isic_4digit` and `unit_isic_4digit` benchmarks", {
-  co2 <- tibble(
-    activity_uuid_product_uuid = c("a", "a", "a"),
-    co2_footprint = c(2, 3, 3),
-    ei_activity_name = c("a", "a", "a"),
-    ei_geography = c("a", "a", "a"),
-    isic_4digit = c("'12'", "'123'", "'1234'"),
-    tilt_sector = c("a", "a", "a"),
-    tilt_subsector = c("a", "a", "a"),
-    unit = c("a", "a", "a")
-  )
-  isic <- emissions_profile_any_compute_profile_ranking(co2) |>
-    filter((str_length(isic_4digit) %in% c(4, 5)) & !(grouped_by %in% c("isic_4digit", "unit_isic_4digit")))
-  expect_false(any(is.na(isic$profile_ranking)))
+test_that("yields `NA` in `profile_ranking` where `tilt_sector` is `NA` and `grouped_by` matches *tilt_sector", {
+  pattern <- aka("tsector")
+  co2 <- example_products(!!pattern := c(NA_character_, "a"))
+
+  out <- emissions_profile_any_compute_profile_ranking(co2)
+
+  should_be_na <- out |>
+    filter(is.na(get_column(out, pattern))) |>
+    filter(grepl(pattern, grouped_by))
+  expect_equal(unique(should_be_na$profile_ranking), NA_integer_)
 })
 
-test_that("more than one highest co2 value should be ranked 1.0", {
-  co2 <- tibble(
-    activity_uuid_product_uuid = c("a", "a", "a"),
-    co2_footprint = c(2, 3, 3),
-    ei_activity_name = c("a", "a", "b"),
-    ei_geography = c("a", "a", "a"),
-    isic_4digit = c("'1234'", "'1234'", "'1234'"),
-    tilt_sector = c("a", "a", "a"),
-    tilt_subsector = c("a", "a", "a"),
-    unit = c("a", "a", "a")
-  )
-  out <- emissions_profile_any_compute_profile_ranking(co2) |>
-    filter(co2_footprint == max(co2_footprint))
-  expect_equal(unique(out$profile_ranking), 1.0)
+test_that("yields `NA` in `profile_ranking` where `*isic_4digit` has 2-3 digits and `grouped_by` matches *isic_4digit", {
+  name <- "input_isic_4digit"
+  co2 <- example_inputs(!!name := c("'12'", "'123'", "'1234'"))
+
+  out <- emissions_profile_any_compute_profile_ranking(co2)
+
+  isic_has_2_or_3_digits_plus_quotes <- str_length(out[[name]]) %in% c(4, 5)
+  relevant_benchamrks <- grepl(name, out$grouped_by)
+  special_cases <- isic_has_2_or_3_digits_plus_quotes & relevant_benchamrks
+
+  expect_true(all(is.na(filter(out, special_cases)$profile_ranking)))
+  # In all other cases `profile_ranking` should not be NA
+  expect_false(any(is.na(filter(out, !special_cases)$profile_ranking)))
```

Comment:
@maurolepore Have you also tested this with `name <- "isic_4digit"` ? It seems it works with `isic_4digit`, however I just want you to check this again. Thanks!